Haptics/app/src/main/java/com/example/android/haptics/samples/ui/views/HomeViewModel.kt

```diff
@@ -0,0 +1,257 @@
+        fun provideFactory(
+            application: Application,
+            scaffoldState: ScaffoldState,
+            scrollState: ScrollState,
+        ): ViewModelProvider.Factory = object : ViewModelProvider.Factory {
+            @Suppress("UNCHECKED_CAST")
+            override fun <T : ViewModel> create(modelClass: Class<T>): T {
+                val androidRorLater = Build.VERSION.SDK_INT >= Build.VERSION_CODES.R // API Level 30.
+                val vibrator = application.getSystemService(Vibrator::class.java)
+
+                // This state object is responsible for providing all the information necessary
+                // for the UI to build the home page of buttons, separated into categories.
+                val viewModelState = HomeViewModelState(
+                    hapticFeedbackCategories = listOf(
+                        // List of predefined effects.
+                        HapticFeedbackCategory(
+                            application.getString(R.string.home_screen_predefined_effects),
+                            categoryType = HapticFeedbackCategoryType.PREDEFINED_EFFECTS,
+                            buttons = listOf(
+                                HapticFeedbackButton(
+                                    label = application.getString(R.string.home_screen_tick),
+                                    worksOnUserDevice = isEffectSupported(
+                                        vibrator,
+                                        VibrationEffect.EFFECT_TICK
+                                    ),
+                                    vibrationEffectConstant = VibrationEffect.EFFECT_TICK
+                                ),
+                                HapticFeedbackButton(
+                                    application.getString(R.string.home_screen_click),
+                                    isEffectSupported(
+                                        vibrator,
+                                        VibrationEffect.EFFECT_CLICK
+                                    ),
+                                    VibrationEffect.EFFECT_CLICK
+                                ),
+                                HapticFeedbackButton(
+                                    application.getString(R.string.home_screen_heavy_click),
+                                    isEffectSupported(vibrator, VibrationEffect.EFFECT_HEAVY_CLICK),
+                                    VibrationEffect.EFFECT_HEAVY_CLICK
+                                ),
+                                HapticFeedbackButton(
+                                    application.getString(R.string.home_screen_double_click),
+                                    isEffectSupported(vibrator, VibrationEffect.EFFECT_DOUBLE_CLICK),
+                                    VibrationEffect.EFFECT_DOUBLE_CLICK
+                                ),
+                            )
+                        ),
+                        // List of haptic feedback constants.
```

Comment:
Did we only want these two Haptic constants or was the intention to have them all and they just weren't in the mocks?

Reply:
Good question, we might circle this back with UX. These seemed like the most interesting ones in order to keep the screen simple and not overwhelming. The others are also integrated in different UX experiences, so having a button to play it might not be as helpful. My suggestion would be to keep only these two, but it's worth confirming with UX.